src/mail/view/MailViewer.js

```diff
@@ -835,10 +852,16 @@ export class MailViewer {
 
 	/** @return list of inline referenced cid */
 	_loadMailBody(mail: Mail): Promise<Array<string>> {
-		return this._entityClient.load(MailBodyTypeRef, mail.body).then(body => {
+		return Promise.all([
+			this._entityClient.load(MailBodyTypeRef, mail.body),
+			worker.isAllowedExternalSender(mail.sender.address)
```

Comment:
Do you check authentication status? It's not great if it's fake sender and we load external content.